### PR TITLE
initial implementation of treeNetwork

### PR DESCRIPTION
**Pull request by cjyetman**
initial implementation of treeNetwork issue #171

### CODE REVIEW

**Comment by cjyetman** [comment]
need to do minimal documentation and figure out other warnings and errors before merging

File: R/as.treenetdf.R [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  
+  df <- 
+    data.frame(
+      nodeId = 1:(length(data$height) + length(data$labels)),
+      parentId = c(clustparents, leaveparents),
+      name = c(rep('', length(data$height)), data$labels),
+      height = c(data$height, rep(0, length(data$labels)))
+    )
+  
+  if (require('tibble')) { return(tibble::as.tibble(df)) }
```
Comment (christophergandrud) [inline R/as.treenetdf.R]: Is this just to return a data frame if the user doesn't have tibble installed? If so, maybe we should import tibble and just use that.

**Comment by cjyetman** [comment]
Yeah, I didn't want to force a tibble installation just for that, but if you think it's ok....

**Comment by christophergandrud** [comment]
Thanks. We should move to working with the tibble data structures for the other graph types. E.g. with `igraph_to_networkD3`

**Comment by christophergandrud** [comment]
Though, before we roll out tibble for all functions, I'll leave it in as a suggests. Doing this now. . .

**Comment by christophergandrud** [comment]
Did some more doc cleaning and merged it into master.

**Pull request merged**